# Trial Name: rwa67_summer2023.yaml
# ARIAC2023

time_limit: -1 # options: -1 (no time limit) or number of seconds
gazebo_state_logging: false # options: true or false


kitting_trays: # Which kitting trays will be spawn
  tray_ids: [3, 3, 5] 
  slots: [1, 3, 5]

parts:
  bins: # bin params - 8 total bins each bin has nine total slots (1-9)
    bin1:
      - type: 'battery'
        color: 'purple'
        slots: [4,5,6]
    bin2:
      - type: 'regulator'
        color: 'orange'
        slots: [4, 6]
      - type: 'sensor'
        color: 'green'
        slots: [5]
orders:
  - id: 'KITTING1'
    type: 'kitting'
    announcement:
      time_condition: 0
    priority: false
    kitting_task:
      agv_number: 1
      tray_id: 3
      destination: 'warehouse'
      products:
        - type: 'battery'
          color: 'purple'
          quadrant: 1
        - type: 'regulator'
          color: 'orange'
          quadrant: 4
  - id: 'KITTING2'
    type: 'kitting'
    announcement:
      part_place_condition:
        agv: 1
        type: 'battery'
        color: 'purple'
    priority: false
    kitting_task:
      agv_number: 2
      tray_id: 3
      destination: 'warehouse'
      products:
        - type: 'sensor'
          color: 'green'
          quadrant: 2
        - type: 'sensor'
          color: 'green'
          quadrant: 3
  - id: 'KITTING3'
    type: 'kitting'
    announcement:
      part_place_condition:
        agv: 1
        type: 'battery'
        color: 'purple'
    priority: false
    kitting_task:
      agv_number: 3
      tray_id: 5
      destination: 'warehouse'
      products:
        - type: 'battery'
          color: 'purple'
          quadrant: 1
        - type: 'regulator'
          color: 'orange'
          quadrant: 2

challenges:
  - faulty_part:
      order_id: 'KITTING1'
      quadrant1: true